crates/hir-def/src/expr_store.rs

```diff
@@ -320,7 +471,8 @@ impl ExpressionStore {
     }
 
     pub fn is_binding_upvar(&self, binding: BindingId, relative_to: ExprId) -> bool {
-        match self.binding_owners.get(&binding) {
+        let Some(expr_only) = &self.expr_only else { return false };
```

Comment:
I can't tell if this should return `true` or `false`, the tests pass either way 😅 .

Reply:
It's unreachable, it could as well panic. But just to be safe...

Reply:
Should have a `never!`, then?

Reply:
I was too lazy to add it (there's bunch of other places).